engine/jobmaster/dm/api.go

```diff
@@ -67,25 +69,25 @@ func (jm *JobMaster) QueryJobStatus(ctx context.Context, tasks []string) (*JobSt
 
 	var (
 		workerStatusMap = jm.workerManager.WorkerStatus()
-		wg              sync.WaitGroup
 		mu              sync.Mutex
+		wg, ctx2        = errgroup.WithContext(ctx)
 		jobStatus       = &JobStatus{
 			JobID:      jm.ID(),
 			TaskStatus: make(map[string]TaskStatus),
 		}
+		unitState *metadata.UnitState
```

Comment:
we can read `unitState` before/after the concurrent task actions so no lock is needed.

Reply:
fixed https://github.com/pingcap/tiflow/pull/7507/commits/c7db2e8582e32c4d8e057e6d3808f18b83797c9a